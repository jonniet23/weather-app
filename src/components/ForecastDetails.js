import React from "react";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>

      <div className="forecast-details__max-temperature">
        Max {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__min-temperature">
        Min {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__humidity">Humidity {humidity}% </div>

      <div className="forecast-details__wind-speed">
        Wind Speed {wind.speed}mph{" "}
      </div>
      <div className="forecast-details__wind-direction">
        Wind Direction {wind.direction}{" "}
      </div>
    </div>
  );
}
export default ForecastDetails;
