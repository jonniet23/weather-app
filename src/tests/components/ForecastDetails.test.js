import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validprops = {
    date: 1111111,
    humidity: "30",
    wind: {
      speed: 13,
      direction: "s",
    },
    temperature: {
      min: 4,
      max: 11,
    },
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastDetails
        date={validprops.date}
        humidity={validprops.humidity}
        wind={validprops.wind}
        temperature={validprops.temperature}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastDetails
        date={validprops.date}
        humidity={validprops.humidity}
        wind={validprops.wind}
        temperature={validprops.temperature}
      />,
    );

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details__date");
    expect(getByText("30")).toHaveClass("forecast-details__humidity");
    expect(getByTestId("forecast-details")).toHaveClass("forecast-details");
    expect(getByText("11Â°C")).toHaveClass("forecast-details__max-temperature");
  });
});
