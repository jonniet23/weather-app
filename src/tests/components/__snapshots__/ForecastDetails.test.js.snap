// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__max-temperature"
    >
      11°C
    </div>
    <div
      class="forecast-details__min-temperature"
    >
      4°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      30 
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      13 
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      s 
    </div>
  </div>
</DocumentFragment>
`;
